internal/rsl/rsl.go

```diff
@@ -506,6 +506,51 @@ func GetFirstEntry(repo *git.Repository) (*ReferenceEntry, []*AnnotationEntry, e
 	return firstEntry, annotations, nil
 }
 
+// GetFirstEntryForRef returns the very first entry in the RSL for a ref. It is expected to be
+// a reference entry as the first entry in the RSL cannot be an annotation.
+func GetFirstEntryForRef(repo *git.Repository, targetRef string) (*ReferenceEntry, []*AnnotationEntry, error) {
```

Comment:
I think there's a lot of repetition with `GetFirstEntry` except for the single targetRef check. Can you update this so `GetFirstEntry` calls this one with `targetRef=""`, and only do the `targetRef == entry.RefName` check `if targetRef != ""`?

Reply:
Fixed